### ISSUE TITLE
Lava Created files not editable/saving changes

### ISSUE THREAD

**Issue by 4everGM** [Praxxian/lava-flow#64]
Noticed that any changes I make in Foundry are not saved, but only for journals that were imported from Obsidian.  It looks like I can edit/save if I set the sheet configuration to Markdown.  I don't believe it behaved this way before, THINKING it is a v12 issue. A warning error in the console shows this:

![image](https://github.com/user-attachments/assets/b2248cc4-0cb8-4cad-b44f-8505fbb1b934)

Trying:
Host(s):  both Forge and Sqyre
Version V12 Build 331
System:  All I've tested.

**Comment by DaGwann** [comment]
I was losing my mind over this!! Thank you for pointing out that I can edit the journal entries with the sheet as Markdown. But this issue remains, nevertheless.

It's a shame that the module seems abandoned, because it's such a great tool. We can only hope that some attention is given again to this project.

**Comment by Praxxian** [comment]
Yeah I don't remember this being an issue in v11, but it will work if you set the sheet to markdown as @DaGwann said. Kinda annoying. However, your edits in Foundry can be overwritten by re-importing your vault. So it is a somewhat questionable workflow to begin with. That might be why the issue was not really noticed before. I know there is a tool for converting Foundry back to Obsidian, so maybe people are using that so they can update their notes within Foundry and sync everything back up.

I could keep this open to see if I can default imported journals to use the markdown editor, though. At least that will be a little less tedious.

**Comment by shadows-eye** [comment]
I hope you find a solution, as Lava flow does not solve all cases, as I also embed Items into the imported Journals and Actors, etc. With Markdown I can´t edit in Foundry anymore. I love this module hopefully you can fix it.

**Comment by shadows-eye** [comment]
I made some tests, when editing the JSON and removing what is in the markdown part, it now works as expected. So here the not working json ```
{
  "name": "Topic",
  "folder": "FYRds1fjEu7gH0kK",
  "pages": [
    {
      "name": "Kapitel 1 Liebe auf dem Dererund",
      "text": {
        "markdown": "*»Content in Markdown",
        "format": 2,
        "content": "<p>Content...</p>"
      }
``` and the working json extract 
```
{
  "name": "Topic",
  "folder": "FYRds1fjEu7gH0kK",
  "pages": [
    {
      "name": "Kapitel 1 Liebe auf dem Dererund",
      "text": {
        "markdown": "",
        "format": 1,
        "content": "<p>Content...</p>"
      }
```
I made a macro to clean that up, hence remove all in the markdown and put in format 1 as value. This works in the dark eye system but could be a good starting point for a fix.


```
/**
 * Macro: Update Journal Pages in Main Folder + Subfolder Children
 * ---------------------------------------------------------------
 * 1) Lets you pick a "JournalEntry"-type folder (the "main folder").
 * 2) Updates:
 *    - All journal pages in the main folder's direct journals (mainFolder.contents).
 *    - All journal pages in mainFolder.sheet.object.children[0].entries (the subfolder journals).
 * 3) For each journal, we call updateEmbeddedDocuments("JournalEntryPage", [...]) to set:
 *    - text.markdown = ""
 *    - text.format = 1
 *
 * Foundry v10+ stores text in "pages" (JournalEntryPage), not directly on the Journal document.
 */

(async () => {
  // 1) Gather all folders of type "JournalEntry"
  const journalFolders = game.folders.filter(f => f.type === "JournalEntry");
  if (!journalFolders.length) {
    ui.notifications.error("No JournalEntry folders found!");
    return;
  }

  // Build dropdown of those folders
  const folderOptions = journalFolders
    .map(f => `<option value="${f.id}">${f.name}</option>`)
    .join("");

  // Dialog content
  const content = `
    <form>
      <div class="form-group">
        <label for="folder-select">Select a Main Journal Folder:</label>
        <select id="folder-select">
          ${folderOptions}
        </select>
      </div>
    </form>
  `;

  // Show a dialog to pick the folder
  new Dialog({
    title: "Update Journal Pages (Markdown & Format)",
    content,
    buttons: {
      confirm: {
        label: "Update",
        callback: async (html) => {
          // Get the chosen folder ID
          const folderId = html.find("#folder-select").val();
          if (!folderId) {
            ui.notifications.error("No folder selected!");
            return;
          }

          // Retrieve the chosen folder object
          const mainFolder = game.folders.get(folderId);
          if (!mainFolder) {
            ui.notifications.error("Folder not found!");
            return;
          }

          let totalPageUpdates = 0;  // track how many pages we actually updated

          // 2) Update all journals in the main folder
          const mainFolderJournals = mainFolder.contents || [];
          for (const j of mainFolderJournals) {
            // Gather all pages, build update data
            const updates = j.pages.map(page => {
              return {
                _id: page.id,
                "text.markdown": "",
                "text.format": 1
              };
            });
            // Perform the update for all pages
            if (updates.length > 0) {
              await j.updateEmbeddedDocuments("JournalEntryPage", updates).catch(err => {
                console.error(`Failed to update pages for "${j.name}"`, err);
              });
              totalPageUpdates += updates.length;
            }
          }

          // 3) Also update any journals found in mainFolder.sheet.object.children[0].entries
          const childArray = mainFolder.sheet?.object?.children || [];
          const firstChild = childArray[0];
          if (!firstChild) {
            // If there's no child data at all, just notify about main folder updates
            ui.notifications.info(
              `Updated ${totalPageUpdates} page(s) in the main folder. No subfolder child data found.`
            );
            return;
          }

          // The subfolder journals
          const subfolderJournals = firstChild.entries || [];
          for (const doc of subfolderJournals) {
            // Each "doc" is presumably a JournalEntry
            const updates = doc.pages.map(page => {
              return {
                _id: page.id,
                "text.markdown": "",
                "text.format": 1
              };
            });
            if (updates.length > 0) {
              await doc.updateEmbeddedDocuments("JournalEntryPage", updates).catch(err => {
                console.error(`Failed to update pages for "${doc.name}"`, err);
              });
              totalPageUpdates += updates.length;
            }
          }

          // Final notification
          ui.notifications.info(
            `Updated ${totalPageUpdates} page(s) across main folder + subfolder entries.`
          );
        }
      }
    },
    default: "confirm"
  }).render(true);
})();
```

**Comment by shadows-eye** [comment]
I made a macro to clean that up, hence remove all in the markdown and put in format 1 as value. This works in the dark eye system but could be a good starting point for a fix.

``` /**
 * Macro: Update Journal Pages in Main Folder + Subfolder Children
 * ---------------------------------------------------------------
 * 1) Lets you pick a "JournalEntry"-type folder (the "main folder").
 * 2) Updates:
 *    - All journal pages in the main folder's direct journals (mainFolder.contents).
 *    - All journal pages in mainFolder.sheet.object.children[0].entries (the subfolder journals).
 * 3) For each journal, we call updateEmbeddedDocuments("JournalEntryPage", [...]) to set:
 *    - text.markdown = ""
 *    - text.format = 1
 *
 * Foundry v10+ stores text in "pages" (JournalEntryPage), not directly on the Journal document.
 */

(async () => {
  // 1) Gather all folders of type "JournalEntry"
  const journalFolders = game.folders.filter(f => f.type === "JournalEntry");
  if (!journalFolders.length) {
    ui.notifications.error("No JournalEntry folders found!");
    return;
  }

  // Build dropdown of those folders
  const folderOptions = journalFolders
    .map(f => `<option value="${f.id}">${f.name}</option>`)
    .join("");

  // Dialog content
  const content = `
    <form>
      <div class="form-group">
        <label for="folder-select">Select a Main Journal Folder:</label>
        <select id="folder-select">
          ${folderOptions}
        </select>
      </div>
    </form>
  `;

  // Show a dialog to pick the folder
  new Dialog({
    title: "Update Journal Pages (Markdown & Format)",
    content,
    buttons: {
      confirm: {
        label: "Update",
        callback: async (html) => {
          // Get the chosen folder ID
          const folderId = html.find("#folder-select").val();
          if (!folderId) {
            ui.notifications.error("No folder selected!");
            return;
          }

          // Retrieve the chosen folder object
          const mainFolder = game.folders.get(folderId);
          if (!mainFolder) {
            ui.notifications.error("Folder not found!");
            return;
          }

          let totalPageUpdates = 0;  // track how many pages we actually updated

          // 2) Update all journals in the main folder
          const mainFolderJournals = mainFolder.contents || [];
          for (const j of mainFolderJournals) {
            // Gather all pages, build update data
            const updates = j.pages.map(page => {
              return {
                _id: page.id,
                "text.markdown": "",
                "text.format": 1
              };
            });
            // Perform the update for all pages
            if (updates.length > 0) {
              await j.updateEmbeddedDocuments("JournalEntryPage", updates).catch(err => {
                console.error(`Failed to update pages for "${j.name}"`, err);
              });
              totalPageUpdates += updates.length;
            }
          }

          // 3) Also update any journals found in mainFolder.sheet.object.children[0].entries
          const childArray = mainFolder.sheet?.object?.children || [];
          const firstChild = childArray[0];
          if (!firstChild) {
            // If there's no child data at all, just notify about main folder updates
            ui.notifications.info(
              `Updated ${totalPageUpdates} page(s) in the main folder. No subfolder child data found.`
            );
            return;
          }

          // The subfolder journals
          const subfolderJournals = firstChild.entries || [];
          for (const doc of subfolderJournals) {
            // Each "doc" is presumably a JournalEntry
            const updates = doc.pages.map(page => {
              return {
                _id: page.id,
                "text.markdown": "",
                "text.format": 1
              };
            });
            if (updates.length > 0) {
              await doc.updateEmbeddedDocuments("JournalEntryPage", updates).catch(err => {
                console.error(`Failed to update pages for "${doc.name}"`, err);
              });
              totalPageUpdates += updates.length;
            }
          }

          // Final notification
          ui.notifications.info(
            `Updated ${totalPageUpdates} page(s) across main folder + subfolder entries.`
          );
        }
      }
    },
    default: "confirm"
  }).render(true);
})(); ```

**Comment by Praxxian** [comment]
There is an alternative which is to set the sheet type as Markdown during the import, which will then cause the editor to work in markdown and save normally. This seems ideal since Obsidian users are used to working in markdown.

However, it would be good to give users the option to use either MD or HTML. So I'll see about making that a new option.

**Comment by Praxxian** [comment]
Fixed in 4.1.0. There is a new option for using TinyMCE, which is disabled by default in favor of using Markdown.
